### PR TITLE
adding unit tests for labelsets, metrics, and fingerprints types

### PR DESCRIPTION
**Pull request by tariq1890**
Signed-off-by: tariqibrahim <tariq181290@gmail.com>

### CODE REVIEW

File: model/fingerprinting_test.go [hunk 1]
```diff
@@ -0,0 +1,214 @@
+// Copyright 2013 The Prometheus Authors
```
Comment (brian-brazil) [inline model/fingerprinting_test.go]: 2019

File: model/fingerprinting_test.go [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+
+	for i, f := range fingerPrints {
+		if f != expected[i] {
+			t.Errorf("expected Fingerprint %d, but got %d for index %d", expected[i], f, i)
+		}
+	}
+}
+
+func TestFingerprintSet(t *testing.T) {
+	//sets of unequal length
```
Comment (brian-brazil) [inline model/fingerprinting_test.go]: Comments should start with a capital letter and end with a full stop

File: model/labelset_test.go [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+	// valid LabelSet JSON
+	labelSetJSON := `{
+	"labelSet": {
+		"monitor": "codelab",
+		"foo": "bar"
+	}
+}`
+	var c testConfig
+	err := json.Unmarshal([]byte(labelSetJSON), &c)
```
Comment (brian-brazil) [inline model/labelset_test.go]: Hmm, it this code dead yet?

File: model/fingerprinting_test.go (hunk 1 above)
Comment (tariq1890) [reply model/fingerprinting_test.go]: Done

File: model/fingerprinting_test.go (hunk 2 above)
Comment (tariq1890) [reply model/fingerprinting_test.go]: Done

File: model/labelset_test.go (hunk 3 above)
Comment (tariq1890) [reply model/labelset_test.go]: I am not sure I follow :).

Comment (brian-brazil) [reply model/labelset_test.go]: I'm wondering if we still have code that uses this, and if not if we can delete it.

Comment (tariq1890) [reply model/labelset_test.go]: I'll update the year in this file's license header

Comment (tariq1890) [reply model/labelset_test.go]: Are you referring to the JSON Unmarshal specifically ? LabelSet is being used widely.

**Comment by brian-brazil** [comment]
Thanks!

**Pull request merged**